vfio/src/vfio_device.rs

```diff
@@ -682,6 +684,53 @@ impl VfioDevice {
     ///
     /// * `irq_index` - The type (INTX, MSI or MSI-X) of interrupts to disable.
     pub fn disable_irq(&self, irq_index: u32) -> Result<()> {
+        // Mask the interrupts before disabling them. This is required to
+        // prevent spurious interrupt from being triggered after the eventfds
+        // have been unassigned.
+        self.mask_irq(irq_index)?;
```

Comment:
In fact, VFIO kernel code doesn't support mask/unmask msi/msix interrupts, a piece of code from `vfio_pci_intrs.c` as below:  
```C
int vfio_pci_set_irqs_ioctl(struct vfio_pci_device *vdev, uint32_t flags,
			    unsigned index, unsigned start, unsigned count,
			    void *data)
{
	int (*func)(struct vfio_pci_device *vdev, unsigned index,
		    unsigned start, unsigned count, uint32_t flags,
		    void *data) = NULL;

	switch (index) {
	...
		case VFIO_PCI_MSI_IRQ_INDEX:
	case VFIO_PCI_MSIX_IRQ_INDEX:
		switch (flags & VFIO_IRQ_SET_ACTION_TYPE_MASK) {
		case VFIO_IRQ_SET_ACTION_MASK:
		case VFIO_IRQ_SET_ACTION_UNMASK:
			/* XXX Need masking support exported */
			break;
		case VFIO_IRQ_SET_ACTION_TRIGGER:
			func = vfio_pci_set_msi_trigger;
			break;
		}
		break;
	...
	}

	if (!func)
		return -ENOTTY;

	return func(vdev, index, start, count, flags, data);
}


```

Reply:
I checked the kernel code and I agree it should error out. I'm very confused now since this PR really fixes the issue I ran into with the MSI NIC card I have on my system.

Reply:
Oh I understand why it fixed the issue, that's because it fails to `MASK` :(